// angular-highlight-js/dist/esm/src/highlightJs.module.js
import { NgModule } from '@angular/core';
import { HighlightJsDirective } from './highlightJs.directive';
import { HIGHLIGHT_JS } from './highlightJs.token';
export var HighlightJsModule = (function () {
    function HighlightJsModule() {
    }
    HighlightJsModule.forRoot = function (hljs) {
        return {
            ngModule: HighlightJsModule,
            providers: [{
                    provide: HIGHLIGHT_JS,
                    // workaround a bug with angular where circular objects can't be used as DI tokens
                    // tracking issue: https://github.com/angular/angular/issues/13685
                    useValue: { highlight: function (language, source) { return hljs.highlight(language, source); } }
                }]
        };
    };
    HighlightJsModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective]
                },] },
    ];
    /** @nocollapse */
    HighlightJsModule.ctorParameters = function () { return []; };
    return HighlightJsModule;
}());
